### PR TITLE
Added docking feature for differential drive robot

### PR DESCRIPTION
**Pull request by padhupradheep**
Now the robot can undock using the newly created service.

### CODE REVIEW

File: configs/auto_docking/auto_docking.yaml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 auto_docking:
+  differential_drive: True
+  max_vel_limit: 0.035 # Irrespective of the magnitude for the final stage of docking. 
+  docking_pose: 0.16 # WARNING: PLEASE USE THE COMPANY SPECIFIED VALUE
+  undocking_pose: 0.5 # This is the minimum safest distance for the robot to perform operation outside the safety region of the docking station. 
   kp:
-    x:           0.3
+    # x:           0.3
+    x:           0.6 # Differential drive - CAUTION: PLEASE DO NOT GIVE HIGH VALUES. It might make the robot go at very high speed.
```
Comment (madMAx43v3r) [inline configs/auto_docking/auto_docking.yaml]: That warning is obsolete now right?

Comment (madMAx43v3r) [reply configs/auto_docking/auto_docking.yaml]: Also can remove the commented out line above.

File: configs/auto_docking/auto_docking_sim.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 auto_docking:
+  differential_drive: True
   kp:
-    x:           0.3
-    y:           1.5
+    # x:           0.3
+    x:           0.6
```
Comment (madMAx43v3r) [inline configs/auto_docking/auto_docking_sim.yaml:5]: Here you replaced the y by x. So now there is no y setting.

Comment (padhupradheep) [reply configs/auto_docking/auto_docking_sim.yaml:5]: This is for simulation, the entire module for simulation needs to be adjusted. I will do it over the weekend or maybe next week.

**Pull request merged**